### PR TITLE
S108 yaml rules

### PR DESCRIPTION
**Pull request by eschultink**
### Features
 - [connector rules to JSON or something](https://app.asana.com/0/1201039336231823/1201118215070899/f) - wip; idea is that if deployed with `rules.yaml` file, will read that in lieu of choosing java class - allowing customers additional control. 

To do:
 - [x] figure out how to test locally with a file in the file system - actually more straightforward how to do it in production, with JAR deployment (as just add it to `deployment/` directory)

### Change implications

 - dependencies added/changed? **yes** - Jackson yaml support

### CODE REVIEW

File: java/core/src/test/java/co/worklytics/psoxy/RulesTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @SneakyThrows
+    @Test
+    public void yaml_file() {
+        byte[] data = TestUtils.getData("rules/example.yaml");
+
+        ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+
+        Rules fromYaml = objectMapper.readerFor(Rules.class)
```
Comment (jlorper) [inline java/core/src/test/java/co/worklytics/psoxy/RulesTest.java:64]: this test is basically the same: is not testing any rule but just that the object mapper can read from bytes as opposed to String like in the first test.

File: java/impl/gcp/src/main/java/co/worklytics/psoxy/Route.java [hunk 2]
```diff
@@ -51,10 +55,13 @@
      * use should be overridden with something
      */
     static final String DEFAULT_SALT = "salt";
+    static final String PATH_TO_RULES_FILES = "/rules.yaml";
 
-    ConfigService config;
+    @Getter
+    ConfigService config  = new EnvVarsConfigService();;
```
Comment (jlorper) [inline java/impl/gcp/src/main/java/co/worklytics/psoxy/Route.java:61]: extra semicolon
```suggestion
    ConfigService config  = new EnvVarsConfigService();
```

**Review by jlorper — APPROVED** [review]
A couple minor things

File: java/core/src/test/java/co/worklytics/psoxy/RulesTest.java (hunk 1 above)
Comment (eschultink) [reply java/core/src/test/java/co/worklytics/psoxy/RulesTest.java:64]: yeah, and just that no encoding issues cause weirdness with the regexes.

**Pull request merged**